src/code/containers/challenge-container-selector.js

```diff
@@ -43,27 +44,33 @@ class ChallengeContainerSelector extends Component {
   }
 
   componentWillMount() {
-    const { routeParams, navigateToCurrentRoute, navigateToChallenge } = this.props;
-    if (hasChangedRouteParams(this.props)) {
+    const { navigateToCurrentRoute, navigateToChallenge, authoring } = this.props;
+    // the URL's challengeId is only used for initial routing, so prioritize the numeric route params
```

Comment:
Wow yes, that is so much nicer! I had no idea you could have multiple route patterns like that. This is great, thanks Sam